### PR TITLE
Search Indexing product's secondary categories

### PR DESCRIPTION
**Pull request by drzraf**

### CODE REVIEW

**Comment by drzraf** [comment]
ping?

**Comment by drzraf** [comment]
ping ?

**Comment by drzraf** [comment]
ping?

**Comment by firstred** [comment]
Hi,

Sorry for the long wait (are you still around?)

Could you please rebase this pull request and add a description so that we can have a look at it again? Thanks in advance!

**Comment by drzraf** [comment]
still around...
but still with 1.6.x these days so I'm not in a situation to rebase/test
this effectively.
I don't even know whether this part of the code change a lot or not.

In any case, please feel free to rebase/push.


best regards

**Comment by drzraf** [comment]
rebased against 1.6.1.x + added description.
It applies and works flawlessly. Feel free to merge.
(for the record this is #PSCSX-7006)

Given that 1.7 is not an option for existing shops, there is no point in working on the "develop".

File: classes/Search.php [hunk 1]
```diff
@@ -383,6 +383,20 @@ public static function getTags($db, $id_product, $id_lang)
      * @param int $id_lang
      * @return string
      */
+    public static function getSecondaryCategories($db, $id_product, $id_lang)
+    {
+        $categoriesArray = $db->executeS(sprintf('
```
Comment (aleeks) [inline classes/Search.php]: Why did you construct your query with a sprintf ? o_o

**Comment by drzraf** [comment]
It's sometimes clearer and easier to optionally quote this.
I could change that if you prefer a the line to look like:
> executeS(" xxx " .  _DB_PREFIX_ , " xxx {$id_lang} ...");

**Comment by aleeks** [comment]
ping @antoin-m  @maximebiloe  what do you think ?

File: classes/Search.php (hunk 1 above)
Comment (drzraf) [inline classes/Search.php]: It's sometimes clearer and easier to optionally quote this.
I could change that if you prefer a the line to look like:
> executeS(" xxx " .  _DB_PREFIX_ , " xxx {$id_lang} ...");

**Comment by antoin-m** [comment]
Hey @drzraf, thank you for your contribution!

Unfortunately we are not merging new features into the 1.6.1.x branch. Feel free to add this feature to the 1.7 version of PrestaShop by opening a new PR on the develop branch.

**Comment by drzraf** [comment]
1.7 is a no-go (NO MIGRATION AVAILABLE).
At this point I hugely advise you to accept (reasonable) features into 1.6 unless you intend to kill PSP future.

**Comment by aleeks** [comment]
Hello @drzraf 
PrestaShop migration will be available soon, we are working on this.

**Pull request closed without merging**